geoutils/georaster.py

```diff
@@ -579,6 +579,20 @@ def reproject(self, dst_ref=None, dst_crs=None, dst_size=None, dst_bounds=None,
             dst_data = np.ones(dst_shape)
             reproj_kwargs.update({'destination': dst_data})
 
+        # Check that reprojection is actually needed
+        if all([
+                (dst_transform == self.transform) or (dst_transform is None),
+                (dst_crs == self.crs) or (dst_crs is None),
+                (dst_size == self.shape) or (dst_size is None),
+                (dst_res == self.res) or (dst_res == self.res[0] == self.res[1]) or (dst_res is None)
+        ]):
+            if (nodata == self.nodata) or (nodata is None):
+                warnings.warn("Output projection, bounds and size are identical -> return self")
+                return self
+
+            else:
+                raise ValueError("Only nodata is different, use self.set_ndv")
```

Comment:
or we could default to a warning + apply set_ndv()? not sure this is going to happen very often

Reply:
Actually, one thing is that set_ndv modifies self in place. This needs to return a copy, so that will be the only difference.
Should there be an option in set_ndv to make a copy?? Maybe not, users can always copy then set_ndv if needed.